espconn.h compile error ipv4_addr_t undefined

### Basic Infos

- [x ] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [x ] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [x ] I have tested that the issue is present in current master branch (aka latest git).
- [x ] I have searched the issue tracker for a similar issue.
- [x ] If there is a stack dump, I have decoded it.
- [x ] I have filled out all fields below.

#### Platform

- Hardware:				[ESP-12]
- Core Version:			[V2.5, 11/2/2019]
- Development Env:		[Arduino IDE]
- Operating System:		[UbuntuS]

### Settings in IDE

- Module:  			[Wemos D1 mini r2]
- Flash Mode:			[dio]
- Flash Size: 		[4MB]
- lwip Variant:   [v1.4|Higher Bandwidth]
- Reset Method:		[nodemcu]
- Flash Frequency:	[40Mhz]
- CPU Frequency:		[80Mhz]
- Upload Using:		[SERIAL]
- Upload Speed:   [115200] (serial upload only)


### Problem Description

Include of espconn.h results in an compiler error (worked in V2.4.2):
```
In file included from /tmp/arduino_modified_sketch_278302/WiFiClient.ino:2:0:
/home/mfg/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/tools/sdk/include/espconn.h:451:54: error: 'ipv4_addr_t' has not been declared
 typedef void (*dns_found_callback)(const char *name, ipv4_addr_t *ipaddr, void *callback_arg);
                                                      ^
```
Following addition in espconn.h solves it:
```cpp
#ifndef ipv4_addr_t
#define ipv4_addr_t ip_addr_t
#endif
```

### [MCVE](https://stackoverflow.com/help/mcve) Sketch

```cpp

#include <Arduino.h>
#include <espconn.h>

void setup() {

}

void loop() {

}

```

### Debug Messages

```
Debug messages go here
```

--- a/cores/esp8266/IPAddress.h
+++ b/cores/esp8266/IPAddress.h
@@ -32,6 +32,7 @@
 #define LWIP_IPV6_NUM_ADDRESSES 0
 #define ip_2_ip4(x) (x)
 #define ipv4_addr ip_addr
+#define ipv4_addr_t ip_addr_t
 #define IP_IS_V4_VAL(x) (1)
 #define IP_SET_TYPE_VAL(x,y) do { (void)0; } while (0)
 #define IP_ANY_TYPE (&ip_addr_any)

